pdp/context.go

```diff
@@ -317,3 +317,20 @@ type Evaluable interface {
 	setOrder(ord int)
 	describe() string
 }
+
+// Iterable interface defines all PDP entities having some parent-child relationship
```

Comment:
I don't think that `Iterable` interface is good idea. It's simpler just to add Marshal to Evaluable, Rule, Expression and other involved structures.